File: apps/backend/tests/mock_services.py
from pathlib import Path
from queue import Queue

from src.core.constants import PlatformType
from src.services.auth_service import AuthService
from src.services.login_service import LoginService


class MockAuthService(AuthService):
    """
    测试用的认证服务mock实现
    """

    def __init__(self, cookie_valid: bool = True):
        self.cookie_valid = cookie_valid

    async def cookie_auth_douyin(self, account_file: Path) -> bool:
        return self.cookie_valid

    async def cookie_auth_tencent(self, account_file: Path) -> bool:
        return self.cookie_valid

    async def cookie_auth_ks(self, account_file: Path) -> bool:
        return self.cookie_valid

    async def cookie_auth_xhs(self, account_file: Path) -> bool:
        return self.cookie_valid

    async def cookie_auth_bilibili(self, account_file: Path) -> bool:
        return self.cookie_valid

    async def check_cookie(self, platform_type: int, file_path: str) -> bool:
        # Check if type is a valid PlatformType enum value or int
        try:
            PlatformType(platform_type)
            return self.cookie_valid
        except ValueError:
            return False


class MockLoginService(LoginService):
    """
    测试用的登录服务mock实现
    """

    def __init__(self, login_status: bool = True, cookie_valid: bool = True):
        self.login_status = login_status
        self.cookie_valid = cookie_valid

    async def douyin_cookie_gen(self, user_id: str, status_queue: Queue) -> None:
        await status_queue.put("https://mock-qrcode-url.com/douyin")
        if self.login_status:
            if self.cookie_valid:
                await status_queue.put("200")
            else:
                await status_queue.put("500")
        else:
            await status_queue.put("500")

    async def get_tencent_cookie(self, user_id: str, status_queue: Queue) -> dict:
        await status_queue.put("https://mock-qrcode-url.com/tencent")
        if self.login_status:
            if self.cookie_valid:
                await status_queue.put("200")
            else:
                await status_queue.put("500")
        else:
            await status_queue.put("500")
        return {}

    async def get_ks_cookie(self, user_id: str, status_queue: Queue) -> None:
        await status_queue.put("https://mock-qrcode-url.com/ks")
        if self.login_status:
            if self.cookie_valid:
                await status_queue.put("200")
            else:
                await status_queue.put("500")
        else:
            await status_queue.put("500")

    async def xiaohongshu_cookie_gen(self, user_id: str, status_queue: Queue) -> None:
        await status_queue.put("https://mock-qrcode-url.com/xiaohongshu")
        if self.login_status:
            if self.cookie_valid:
                await status_queue.put("200")
            else:
                await status_queue.put("500")
        else:
            await status_queue.put("500")

    async def bilibili_cookie_gen(self, user_id: str, status_queue: Queue) -> None:
        await status_queue.put("https://mock-qrcode-url.com/bilibili")
        if self.login_status:
            if self.cookie_valid:
                await status_queue.put("200")
            else:
                await status_queue.put("500")
        else:
            await status_queue.put("500")

File: apps/backend-node/tests/mock-services.ts
/**
 * Mock services for testing.
 * Mirrors: apps/backend/tests/mock_services.py
 */

import { EventEmitter } from 'events';
import type { AuthService } from '../src/services/auth-service.js';
import type { CookieService } from '../src/services/cookie-service.js';
import type { LoginService } from '../src/services/login-service.js';

/**
 * MockAuthService — always returns configurable cookie_valid
 */
export class MockAuthService implements AuthService {
    constructor(public cookieValid: boolean = true) { }

    async cookieAuthDouyin(_accountFile: string): Promise<boolean> { return this.cookieValid; }
    async cookieAuthTencent(_accountFile: string): Promise<boolean> { return this.cookieValid; }
    async cookieAuthKs(_accountFile: string): Promise<boolean> { return this.cookieValid; }
    async cookieAuthXhs(_accountFile: string): Promise<boolean> { return this.cookieValid; }
    async cookieAuthBilibili(_accountFile: string): Promise<boolean> { return this.cookieValid; }
}

/**
 * MockCookieService — all methods return configurable cookieValid.
 */
export class MockCookieService implements CookieService {
    constructor(public cookieValid: boolean = true) { }

    async cookieAuthDouyin(_accountFile: string): Promise<boolean> { return this.cookieValid; }
    async cookieAuthTencent(_accountFile: string): Promise<boolean> { return this.cookieValid; }
    async cookieAuthKs(_accountFile: string): Promise<boolean> { return this.cookieValid; }
    async cookieAuthXhs(_accountFile: string): Promise<boolean> { return this.cookieValid; }
    async cookieAuthBilibili(_accountFile: string): Promise<boolean> { return this.cookieValid; }
    async checkCookie(platformType: number, _filePath: string): Promise<boolean> {
        return platformType >= 1 && platformType <= 5 ? this.cookieValid : false;
    }
}

/**
 * MockLoginService — emits mock QR code URL + status code
 */
export class MockTestLoginService implements LoginService {
    loginStatus: boolean;
    cookieValid: boolean;
    pollTimeout: number;
    pollInterval: number;

    constructor(loginStatus = true, cookieValid = true, pollTimeout = 30, pollInterval = 1.0) {
        this.loginStatus = loginStatus;
        this.cookieValid = cookieValid;
        this.pollTimeout = pollTimeout;
        this.pollInterval = pollInterval;
    }

    private emitResult(emitter: EventEmitter, platform: string): void {
        emitter.emit('message', `https://mock-qrcode-url.com/${platform}`);
        emitter.emit('message', this.loginStatus && this.cookieValid ? '200' : '500');
    }

    async douyinCookieGen(_id: string, emitter: EventEmitter, _signal: AbortSignal): Promise<any> { this.emitResult(emitter, 'douyin'); return {}; }
    async getTencentCookie(_id: string, emitter: EventEmitter, _signal: AbortSignal): Promise<any> { this.emitResult(emitter, 'tencent'); return {}; }
    async getKsCookie(_id: string, emitter: EventEmitter, _signal: AbortSignal): Promise<any> { this.emitResult(emitter, 'ks'); return {}; }
    async xiaohongshuCookieGen(_id: string, emitter: EventEmitter, _signal: AbortSignal): Promise<any> { this.emitResult(emitter, 'xiaohongshu'); return {}; }
    async bilibiliCookieGen(_id: string, emitter: EventEmitter, _signal: AbortSignal): Promise<any> { this.emitResult(emitter, 'bilibili'); return {}; }
}
